### PR TITLE
add Traits storage to Protos

### PR DESCRIPTION
**Pull request by sinkingsugar**
Add a way to fetch all traits

### CODE REVIEW

**Comment by alebaffa** [comment]
This will be useful for filtering protos by Trait too. I will refactor after we merge this one.
👍🏻

File: pallets/protos/src/lib.rs [hunk 1]
```diff
@@ -821,79 +838,61 @@ pub mod pallet {
 			if let Some(struct_proto) = <Protos<T>>::get(proto_id) {
 				if let Some(avail) = avail {
 					if avail && struct_proto.include_cost.is_none() {
-						return false
+						return false;
```
Comment (sinkingsugar) [inline pallets/protos/src/lib.rs]: @alebaffa you want a semicolon here (I added in merge)

File: pallets/protos/src/lib.rs [hunk 2]
```diff
@@ -821,79 +838,61 @@ pub mod pallet {
...
 					}
 				}
 
 				if categories.len() == 0 {
-					return Self::filter_tags(tags, &struct_proto)
-				} else {					
-					return Self::filter_category(tags, &struct_proto, categories)
+					return Self::filter_tags(tags, &struct_proto);
```
Comment (sinkingsugar) [inline pallets/protos/src/lib.rs]: @alebaffa also here, if you use `return` use semicolon

File: pallets/protos/src/lib.rs [hunk 3]
```diff
@@ -821,79 +838,61 @@ pub mod pallet {
...
+				} else {
+					return Self::filter_category(tags, &struct_proto, categories);
```
Comment (sinkingsugar) [inline pallets/protos/src/lib.rs]: also here, trick is you could have removed `return` and in that case no need for semicolon

Comment (alebaffa) [reply pallets/protos/src/lib.rs]: Yeah, right! Sorry for this!

**Review by alebaffa — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**